packages/mask/src/plugins/FindTruman/SNSAdaptor/EncryptionCard.tsx

```diff
@@ -1,68 +1,168 @@
+            },
+        },
+        card: {
+            borderRadius: '6px',
+            ':not(:last-child)': {
+                marginBottom: '8px',
+            },
+            display: 'flex',
+            flexDirection: 'column',
+        },
+    }
+})
 
 interface EncryptionCardProps {
-    payload: string
+    clueId: string
 }
 export default function EncryptionCard(props: EncryptionCardProps) {
-    const { payload } = props
+    const { clueId } = props
 
-    const { t } = useI18N()
-    const { address } = useContext(FindTrumanContext)
-    const [failed, setFailed] = useState<boolean>(false)
-    const [message, setMessage] = useState<string>('')
-    const [err, setErr] = useState<{
-        code: number
-        data: any
-    }>()
+    const { classes } = useStyles()
+    const { address, t } = useContext(FindTrumanContext)
+    const [flipped, setFlipped] = useState(false)
+    const [backImgHeight, setBackImgHeight] = useState(0)
 
-    useEffect(() => {
-        setFailed(false)
-        try {
-            const raw = JSON.parse(Buffer.from(payload, 'base64').toString())
-            if (!!raw.cid && !!raw.data) {
-                fetchSecretKey(raw.cid, address)
-                    .then(async (res) => {
-                        setMessage(raw.data)
-                    })
-                    .catch((error) => {
-                        setErr(error)
-                    })
-            }
-        } catch (error) {
-            setFailed(true)
-        }
-    }, [payload])
+    const { value: clue, error } = useAsync(async () => {
+        return clueId
+            ? fetchClue(clueId, address)
+            : { decrypted: false, conditions: undefined, frontImg: '', backImg: '' }
+    }, [clueId, address])
 
     return (
         <CardContent>
-            {!!message && (
+            {clue && (
                 <>
                     <Typography variant="body1" color="text.secondary">
                         {t('plugin_find_truman_decrypted_by')}
                     </Typography>
                     <Divider sx={{ margin: '8px 0' }} />
-                    <Typography variant="h6" color="text.primary">
-                        {message}
-                    </Typography>
+                    <CardContent sx={{ display: 'flex', justifyContent: 'center' }}>
+                        <div style={{ width: 250 }}>
+                            <FlipCard isFlipped={flipped}>
+                                <img
+                                    onLoad={({ target }) => {
+                                        setBackImgHeight((target as HTMLElement).parentElement?.offsetHeight || 0)
+                                    }}
+                                    onClick={({ target }) => {
+                                        setBackImgHeight((target as HTMLElement).parentElement?.offsetHeight || 0)
+                                        backImgHeight && setFlipped(true)
+                                    }}
```

Comment:
use `currentTarget`, then you can omit `as HTMLElement`, because ts will understand currentTarget referrers to `<img />`

```suggestion
                                    onLoad={({ currentTarget }) => {
                                        setBackImgHeight(currentTarget.parentElement?.offsetHeight || 0)
                                    }}
                                    onClick={({ target }) => {
                                        setBackImgHeight(currentTarget.parentElement?.offsetHeight || 0)
                                        backImgHeight && setFlipped(true)
                                    }}
```